icu4c/source/common/rbbiscan.cpp

```diff
@@ -856,6 +856,12 @@ UChar32  RBBIRuleScanner::nextCharLL() {
         return (UChar32)-1;
     }
     ch         = fRB->fRules.char32At(fNextIndex);
+    if (U_IS_SURROGATE(ch)) {
+        RBBIDebugPuts("RBBIRuleScanner::nextCharLL, unpair surrogate");
+        error(U_ILLEGAL_CHAR_FOUND);
+        *fRB->fStatus = U_ILLEGAL_CHAR_FOUND;
```

Comment:
There's no need to set fRB->fStatus here. The error() function does it for you.
Also, could you remove the debug output? Debug output in RBBI is generally either under control of various tracing flags, or if unconditional, is for internal errors that shouldn't be possible. (I admit, this isn't actually described anywhere, but the RBBI code is pretty consistent in following it.)

Reply:
I somehow forget to delete that line. fixed now.